plugins/inputs/modbus/modbus_test.go

```diff
@@ -0,0 +1,261 @@
+			v, _ := t.Value.(uint16)
+			res[0] = res[0] | byte(v<<uint(i))
+		}
+
+		assert.Equal(t, coilTest.ExpectValue, res)
+	}
+}
+
+func TestRegisters(t *testing.T) {
+	serv := mbserver.NewServer()
+	err := serv.ListenTCP("localhost:1502")
+	defer serv.Close()
+	assert.NoError(t, err)
+
+	handler := m.NewTCPClientHandler("localhost:1502")
+	err = handler.Connect()
+	assert.NoError(t, err)
+	defer handler.Close()
+	client := m.NewClient(handler)
+
+	modbus := Modbus{
+		Type:       "TCP",
+		Controller: "localhost",
+		Port:       1502,
+		SlaveId:    1,
+		Timeout:    0,
+		Registers: registers{
+			HoldingRegisters: register{
+				Tags: []tag{
+					{
+						Name:     "Register0",
+						Order:    "AB",
+						DataType: "FLOAT32",
+						Address:  []int{0},
+						Scale:    "/10",
+					},
+					{
+						Name:     "Register1-2",
+						Order:    "ABCD",
+						DataType: "FLOAT32",
+						Address:  []int{1, 2},
+						Scale:    "/1000",
+					},
+					{
+						Name:     "Register3-4",
+						Order:    "ABCD",
+						DataType: "FLOAT32",
+						Address:  []int{3, 4},
+						Scale:    "/10",
+					},
+					{
+						Name:     "Register7",
+						Order:    "AB",
+						DataType: "FLOAT32",
+						Address:  []int{7},
+						Scale:    "/10",
+					},
+					{
+						Name:     "Uint16AB",
+						Order:    "AB",
+						DataType: "UINT16",
+						Address:  []int{10},
+					},
+					{
+						Name:     "Uint16BA",
+						Order:    "BA",
+						DataType: "UINT16",
+						Address:  []int{10},
+					},
+					{
+						Name:     "Int16AB",
+						Order:    "AB",
+						DataType: "INT16",
+						Address:  []int{10},
+					},
+					{
+						Name:     "Int16BA",
+						Order:    "BA",
+						DataType: "INT16",
+						Address:  []int{10},
+					},
+					{
+						Name:     "Int32ABCD",
+						Order:    "ABCD",
+						DataType: "INT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Int32DCBA",
+						Order:    "DCBA",
+						DataType: "INT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Int32BADC",
+						Order:    "BADC",
+						DataType: "INT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Int32CDAB",
+						Order:    "CDAB",
+						DataType: "INT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Uint32ABCD",
+						Order:    "ABCD",
+						DataType: "UINT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Uint32DCBA",
+						Order:    "DCBA",
+						DataType: "UINT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Uint32BADC",
+						Order:    "BADC",
+						DataType: "UINT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "Uint32CDAB",
+						Order:    "CDAB",
+						DataType: "UINT32",
+						Address:  []int{10, 11},
+					},
+					{
+						Name:     "FLOAT32-IEEE",
+						Order:    "ABCD",
+						DataType: "FLOAT32-IEEE",
+						Address:  []int{10, 11},
+					},
+				},
+			},
+		},
+	}
+
+	regTests := []mbTest{}
```

Comment:
Let's see if we can set this up as a [table test](https://github.com/golang/go/wiki/TableDrivenTests).  Each test or subtest should perform one logical check.

Here is an example from within Telegraf:https://github.com/influxdata/telegraf/blob/master/plugins/outputs/influxdb/http_test.go#L61